scripts/make_object_catalog.py

```diff
@@ -0,0 +1,236 @@
+            output_dir,
+            '_'.join((filename_prefix, str(tract), patch.replace(',', ''))) + '.parquet',
+        )
+
+        if not overwrite and (os.path.exists(file_path) or os.path.exists(file_path + '.empty')):
+            if verbose:
+                print("  Skipping tract %d, patch %s because output file exist" % (tract, patch))
+            continue
+
+        butler = _ensure_butler_instance(butler)
+        merged_cat = merge_coadd_forced_src(butler, tract, patch, verbose=verbose, **kwargs)
+
+        if merged_cat is None:
+            if verbose:
+                print("  No entries for tract %d, patch %s" % (tract, patch))
+            open(file_path + '.empty', 'w').close()
+            continue
+
+        merged_cat.to_parquet(
+            file_path,
+            engine=parquet_engine,
+            compression=None,
+            index=False,
+        )
+        del merged_cat
+
+
+def merge_coadd_forced_src(butler, tract, patch, keys_join_on=('id',),
+                           filters='ugrizy', verbose=False, return_pandas=True):
+    """Load patch catalogs.  Return merged catalog across filters.
+
+    butler: Butler object or str
+        Either a Butler object or a filename to the repo
+    tract: int
+        Tract in skymap
+    patch: str
+        Patch in the tract in the skymap
+    keys_join_on: iterable of str
+        Join the catalogs for each filter on these fields
+    filters: iterable of str
+        Filter names to load
+
+    Returns
+    --
+    Pandas DataFrame of patch catalog merged across filters.
+    """
+    butler = _ensure_butler_instance(butler)
+
+    # Define the filters and order in which to sort them.:
+    tract_patch_data_id = {'tract': tract, 'patch': patch}
+    try:
+        ref_table = butler.get(datasetType='deepCoadd_ref',
+                               dataId=tract_patch_data_id)
+    except NoResults as e:
+        if verbose:
+            print("  ", e)
+        return
+
+    ref_table = ref_table.asAstropy()
+    isPrimary = ref_table['detect_isPrimary']
+    if not isPrimary.any():
+        if verbose:
+            print("  No good isPrimary entries for tract {}, patch {}".format(tract, patch))
+        return
+
+    merged_cat = ref_table[isPrimary]
+    del ref_table
+
+    merged_cat['tract'] = int(tract)
+    merged_cat['patch'] = str(patch)
+
+    cat_dtype = None
+    missing_filters = list()
+    for filter_this in filters:
+        filter_name = filters.get(filter_this) if hasattr(filters, 'get') else filter_this
+        this_data_id = dict(tract_patch_data_id, filter=filter_name)
+        try:
+            cat = butler.get(datasetType='deepCoadd_forced_src',
+                             dataId=this_data_id)
+        except NoResults as e:
+            if verbose:
+                print("  ", e)
+            missing_filters.append(filter_this)
+            continue
+
+        cat = cat.asAstropy()
+        cat = cat[isPrimary]
```

Comment:
@wmwv looking at the script again, I am actually confused. Isn't the fact that we can do this line (note that `isPrimary` is from `deepCoadd_ref` and used by all filters) means that we don't need to match by `id` at all? We should be able to just do a `hstack`, since the row order/number must be the same for all filters??

Reply:
@wmwv I've updated the script to use `hstack` instead of `join` and with limited tests I believe it works as expected. If you can confirm that indeed `deepCoadd_forced_src` and `deepCoadd_ref` are expected to have the same row order/number, that would be great.